api/src/AppBundle/Service/OrgService.php

```diff
@@ -210,6 +210,11 @@ public function addFromCasrecRows(array $data)
 
                 $client = $this->upsertClientFromCsv($row, $namedDeputy);
                 if ($client instanceof Client) {
+
+                    if (!$this->clientHasSwitchedOrganisation($client) && $this->clientHasNewNamedDeputy($client, $namedDeputy)) {
+                        $client->setNamedDeputy($namedDeputy);
+                    }
+
```

Comment:
Whilst ensuring that we no longer move Client's into new orgs, we also must ensure that we don't auto update their named deputy either, by only updating deputies for clients who have not moved orgs.

Reply:
Question on this - it looks like the client is always going to have the `$namedDeputy` from the CSV row at this point and it will never be null as its set in `upsertClientFromCsv()` so the second check on `clientHasNewNamedDeputy()` seems redundant? Unless I've missed something and `$namedDeputy` can be null at this point? 🤔

Reply:
I did mull over this too, but for an existing client they won't have hit the setter in `upsertClientFromCsv()` and we've have been stung in the past assuming that all existing clients have a named deputy set. I think technically you are correct, if we remove it we get the same result, except we will be needlessly setting it to it's current value for the majority of them, whereas this is an attempt to only do an update if we actually need to - it's arguable that by being more restrictive we somehow don't update one that is needed, but we also stand a chance of not updating one that we shouldn't. I think it is better to not update than to update as not updating does not have any side effects.

Reply:
Makes sense - better to not have a potential data breach 👍